### ISSUE TITLE
English videos annotated as dubbed in other languages

### ISSUE THREAD

**Issue by radinamatic** [learningequality/ka-lite#4994]
## Summary

Downloaded a few English videos, but after installing a Spanish language pack, I saw that the same videos appear as already downloaded when I switched the default language to Spanish. Once a deleted 2 of them and re-downloaded, they were correct, dubbed version. The same thing repeated when I re-tested with subsequent French language pack - videos appear as downloaded, when played it is clear that it's the English version, so one needs to go back, delete and re-download to get the dubbed version.     

## Branch or installer method

Both Beta2 in Ubuntu & build 176 on Windows. 

## Traceback or relevant snippet from server.log

```
No visible error in the console and nothing of consequence in the server.log.
```

## How to reproduce

1. Download a couple of English videos.
1. Install language pack & switch the default language.
1. Check the Manange > Videos page and you'll see the same videos (test the play to confirm they are in English).
1. Delete and re-download, and you'll get the dubbed version.

**Comment by radinamatic** [comment]
Forgot to ping @aronasorman!

**Comment by radinamatic** [comment]
Plus, there was this funny error of  `0 videos deleted successfully`... :grinning: 
I did, in fact, delete 2 - maybe I should put this in a separate issue.

![ubuntu 14 04 ready for ka lite running - oracle vm virtualbox_289](https://cloud.githubusercontent.com/assets/1457929/13749781/302a2164-ea03-11e5-9627-f4919722e1ed.png)

**Comment by radinamatic** [comment]
> Plus, there was this funny error of 0 videos deleted successfully... :grinning:
I did, in fact, delete 2 - maybe I should put this in a separate issue.

This is actually correct, me thinks, as it's trying to "de-annotate" videos that shouldn't be there in the French videos topic tree... So it didin't in fact, delete anything, right @aronasorman?

Anyway, confirmed it happening in the upgrade scenario 0.14 > 0.16 on XP.

**Comment by radinamatic** [comment]
This is a `release blocker` IMHO.
@aronasorman

**Comment by aronasorman** [comment]
Won't have time. @rtibbles or @MCGallaspy can you look into this issue?

**Comment by rtibbles** [comment]
This is deliberate, and done when subtitles are available, as a fallback.
This was discussed in another issue for which I don't have a reference
right now.

Is it happening when subtitles are not available? Or interfering with
download?

**Comment by radinamatic** [comment]
> This is deliberate, and done when subtitles are available, as a fallback. This was discussed in another issue for which I don't have a reference right now.

Ok, I wasn't aware of that discussion. If you find the reference, it would be great.

> Is it happening when subtitles are not available? Or interfering with download?

I don't recall since I haven't had a lot of chance to view subtitles. It's not interfering with the download (a part from the confusing message `0 videos deleted`), but it gives a wrong impression/information to the user. 

This approach might be a good fallback solution, but we should definitely inform the user that videos that they, at that point, view as present/downloaded in a localized topic tree are, in fact, NOT the dubbed ones. If English videos appear there by default, Admin user must also have an option to re-load the topic tree and see which & how many dubbed videos they have. I don't see "deleting" and re-downloading to get the dubbed version a very good user experience.

**Comment by MCGallaspy** [comment]
Confirmed on 0.16.x, though oddly if you download English videos *after* downloading the Spanish languagepack, the video is*not* marked as available in Spanish.

**Comment by MCGallaspy** [comment]
@rtibbles alarmingly the desired fallback behavior doesn't happen either -- that is, having downloaded an English video which presumably has Spanish subs, the video is *not* marked as available when the default language is changed to Spanish.

**Comment by radinamatic** [comment]
> Then the Spanish video is marked as available (though it plays with English audio).

AFAIK, if it plays "English audio" then it's English video... :smiley:

**Comment by MCGallaspy** [comment]
I mean that the video is marked as available in the Spanish video download UI.

**Comment by radinamatic** [comment]
Yes, sorry, I sometimes can't resist being overly-pedantic. 

On another hand it's worth remembering that most of the non-English KA videos (valid for Spanish, not sure for others) are **not** technically dubbed videos. Meaning that they are not made by simply overlaying the original (English) video with audio track which is a (spoken) translation of original English audio. They are **re-made** (both the video & audio tracks) by following very closely Sal's "script" from the original video. That's the reason why Spanish subs cannot be used on Spanish videos, and have accessibility issues described in #3913. 

Doesn't really help us with this issue, but worth keeping in mind going forward.

**Comment by MCGallaspy** [comment]
@rtibbles I've been looking at this problem all afternoon. Here are my observations/questions:

1. **Observation:** video/topic "availability" on the video download page depends on the `files_complete` and `total_files` attributes.
2. **Question:** my observation leads me to believe that these attributes have "cross talk" between the content databases -- in other words, that `total_files` for the Spanish content database depends on the number of English videos *and* the number of dubbed Spanish videos. Is that true?
3. **Observation:** if it *is* true then these attributes are not IMO a good proxy for "availability" in the sense indicated by Radina. Spanish topic/video "availability" for video download purposes shouldn't depend on the English content database. I spent ~~some~~ way too much time trying to figure out if "availability" could be computed on demand. The problem is that computing this attribute for a topic depends linearly on the # of children for that topic and would increase/dominate the run time. (Whereas currently AFAICT it's dominated by the database lookup, which should be done in constant time.)
4. **Observation:** If it's not true, there might be a bug in how `total_files` and `files_complete` are computed/updated.
5. **Question:** As far as you know, are these attributes used elsewhere in the application?

**Comment by rtibbles** [comment]
Fixed in #5045
